### PR TITLE
Pilot e2e test support for local docker registry

### PR DESCRIPTION
**Pull request by nmittler**
- Various changes to the e2e tests
- Re-introducing old mixer docs (with many changes) for
running a local docker registry in k8s.

### CODE REVIEW

File: tests/util/localregistry/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+$ make docker push HUB=localhost:5000 TAG=latest
+```
+
+### Run with Local Images
+
+#### Running E2E Tests
+If you're running e2e tests, you can set the test flags:
+
+```shell
+$ go test <TEST_PATH> -hub=localhost:5001 -tag=latest
```
Comment (ZackButcher) [inline tests/util/localregistry/README.md]: `localhost:5000`, right?

File: tests/util/localregistry/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+#### Hard-coding the Image URL
+
+You can also modify the image URLs in your deployment yaml files directly:
+
+```yaml
+image: localhost:50001/<APP_NAME>:latest
```
Comment (ZackButcher) [inline tests/util/localregistry/README.md]: 5000 here too, right?

File: tests/util/localregistry/README.md (hunk 1 above)
Comment (nmittler) [reply tests/util/localregistry/README.md]: done.

File: tests/util/localregistry/README.md (hunk 2 above)
Comment (nmittler) [reply tests/util/localregistry/README.md]: done.

**Comment by nmittler** [comment]
@ZackButcher PTAL

**Pull request merged**